src/panels/lovelace/editor/select-dashboard/hui-dialog-select-dashboard.ts

```diff
@@ -0,0 +1,193 @@
+import { mdiClose } from "@mdi/js";
+import type { CSSResultGroup } from "lit";
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, query, state } from "lit/decorators";
+import { fireEvent } from "../../../../common/dom/fire_event";
+import "../../../../components/ha-md-dialog";
+import "../../../../components/ha-dialog-header";
+import "../../../../components/ha-icon-button";
+import "../../../../components/ha-md-select";
+import "../../../../components/ha-md-select-option";
+import "../../../../components/ha-button";
+import "../../../../components/ha-circular-progress";
+import type { LovelaceConfig } from "../../../../data/lovelace/config/types";
+import type { LovelaceDashboard } from "../../../../data/lovelace/dashboard";
+import { fetchDashboards } from "../../../../data/lovelace/dashboard";
+import { haStyleDialog } from "../../../../resources/styles";
+import type { HomeAssistant } from "../../../../types";
+import type { SelectDashboardDialogParams } from "./show-select-dashboard-dialog";
+import type { HaMdDialog } from "../../../../components/ha-md-dialog";
+
+@customElement("hui-dialog-select-dashboard")
+export class HuiDialogSelectDashboard extends LitElement {
+  public hass!: HomeAssistant;
+
+  @state() private _params?: SelectDashboardDialogParams;
+
+  @state() private _dashboards?: LovelaceDashboard[];
+
+  @state() private _fromUrlPath?: string | null;
+
+  @state() private _toUrlPath?: string | null;
+
+  @state() private _config?: LovelaceConfig;
+
+  @state() private _saving = false;
+
+  @query("ha-md-dialog") private _dialog?: HaMdDialog;
+
+  public showDialog(params: SelectDashboardDialogParams): void {
+    this._config = params.lovelaceConfig;
+    this._fromUrlPath = params.urlPath;
+    this._params = params;
+    this._getDashboards();
+  }
+
+  public closeDialog(): void {
+    this._saving = false;
+    this._dashboards = undefined;
+    this._toUrlPath = undefined;
+    this._dialog?.close();
+  }
+
+  private _dialogClosed(): void {
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  protected render() {
+    if (!this._params) {
+      return nothing;
+    }
+
+    const dialogTitle =
+      this._params.header ||
+      this.hass.localize("ui.panel.lovelace.editor.select_dashboard.header");
+
+    return html`
+      <ha-md-dialog
+        open
+        @closed=${this._dialogClosed}
+        .ariaLabel=${dialogTitle}
+        .disableCancelAction=${this._saving}
+      >
+        <ha-dialog-header slot="headline">
+          <ha-icon-button
+            slot="navigationIcon"
+            .label=${this.hass.localize("ui.dialogs.generic.close") ?? "Close"}
+            .path=${mdiClose}
+            @click=${this.closeDialog}
+            .disabled=${this._saving}
+          ></ha-icon-button>
+          <span slot="title" .title=${dialogTitle}> ${dialogTitle} </span>
```

Comment:
```suggestion
          <span slot="title" .title=${dialogTitle}>${dialogTitle}</span>
```
Just a nitpick but the space would push the title right otherwise